R/module_filter_manager.R

```diff
@@ -1,41 +1,95 @@
 #' Manage multiple `FilteredData` objects
 #'
+#' @description
 #' Oversee filter states across the entire application.
 #'
-#' This module observes changes in the filters of each `FilteredData` object
-#' and keeps track of all filters used. A mapping of filters to modules
-#' is kept in the `mapping_matrix` object (which is actually a `data.frame`)
-#' that tracks which filters (rows) are active in which modules (columns).
+#' @section Slices global:
+#' The key role in maintaining the module-specific filter states is played by the `.slicesGlobal`
+#' object. It is a reference class that holds the following fields:
+#' - `all_slices` (`reactiveVal`) - reactive value containing all filters registered in an app.
+#' - `module_slices_api` (`reactiveValues`) - reactive field containing references to each modules'
+#' `FilteredData` object methods. At this moment it is used only in `srv_filter_manager` to display
+#' the filter states in a table combining informations from `all_slices` and from
+#' `FilteredData$get_available_teal_slices()`.
+#'
+#' During a session only new filters are added to `all_slices` unless [`module_snapshot_manager`] is
+#' used to restore previous state. Filters from `all_slices` can be activated or deactivated in a
+#' module which is linked (both ways) by `attr(, "mapping")` so that:
+#' - If module's filter is added or removed in its `FilteredData` object, this information is passed
+#'   to `SlicesGlobal` which updates `attr(, "mapping")` accordingly.
+#' - When mapping changes in a `SlicesGlobal`, filters are set or removed from module's
+#'  `FilteredData`.
+#'
+#' @section Filter manager:
+#' Filter-manager is split into two parts:
+#' 1. `ui/srv_filter_manager_panel` - Called once for the whole app. This module observes changes in
+#' the filters in `slices_global` and displays them in a table utilizing information from `mapping`:
+#'   - &#9989; (`TRUE`) - filter is active in the module
+#'   - &#10060; (`FALSE`) - filter is inactive in the module
+#'   - &#128306; (`NA`) - filter is not available in the module
```

Comment:
This is causing a LaTeX error
https://github.com/insightsengineering/teal/actions/runs/10143639530/job/28045440046#step:41:389

Reply:
https://github.com/insightsengineering/teal/pull/1278